test/testutil.c

```diff
@@ -400,18 +412,17 @@ int test_mem_eq(const char *file, int line, const char *st1, const char *st2,
 
     if (s1 == NULL && s2 == NULL)
         return 1;
-    if (n1 != n2) {
-        test_fail_message(NULL, file, line, "memory",
-                          "size mismatch %s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
-        return 0;
-    }
-    if (s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
+    if (n1 != n2 || s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
+        char *m1 = print_mem_maybe_null(s1, n1, b1);
+        char *m2 = print_mem_maybe_null(s2, n2, b2);
+
         test_fail_message(NULL, file, line, "memory",
-                          "%s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
+                          "%s %s [%"OSSLzu"] == %s %s [%"OSSLzu"]",
```

Comment:
Assert messages read "here: assertion 'x' failed". Two things: a) condition is noun; b) it shows variables' *names*. Messages in question on the other hand show *values* and read "error: 1 == 2 here." This sounds (as already mentioned) as "there was problem because of this condition." It's as counter-intuitive for programmer as for anybody else.